store/auth.go

```diff
@@ -242,7 +237,7 @@ func refreshDischargeMacaroon(discharge string) (string, error) {
 }
 
 // requestStoreDeviceNonce requests a nonce for device authentication against the store.
-func requestStoreDeviceNonce() (string, error) {
+func requestStoreDeviceNonce(deviceNonceAPI string) (string, error) {
```

Comment:
we should start not using API alone for this,   deviceNonceURL ?  deviceNonceEndpoint ?

Reply:
they could even take actual url.URL given that we need to do string in store.go ? gets a bit annoying in tests though

Reply:
I initially wanted to pass a url.URL but the funcs it calls expect strings anyway so left things alone.

I agree about the name. It didn't feel right calling it "URL" so I went with the option that changed as little as possible. However, I prefer "endpoint", and that matches the name in the more general request-related funcs, so I changed it.